index.js

```diff
@@ -190,10 +190,12 @@ module.exports = {
     let addonProvidedConfig = this._getAddonProvidedConfig(config);
     let shouldCompileModules = this._shouldCompileModules(config);
 
-    let providedAnnotation = config['ember-cli-babel'] && config['ember-cli-babel'].annotation;
+    let addonOptions = config['ember-cli-babel'] || {};
+    let babelOptions = config.babel || {};
 
     let options = {
-      annotation: providedAnnotation || `Babel: ${this._parentName()}`
+      annotation: addonOptions.annotation || `Babel: ${this._parentName()}`,
+      sourceMaps: addonOptions.sourceMaps || babelOptions.sourceMaps
```

Comment:
Hmm, I'm not sure that we should do this sort of fallback. Each option needs to have a single place to live, if we have to support both locations one should be deprecated.